scripts/src/sanitycheckpr/sanitycheckpr.py

```diff
@@ -17,23 +23,69 @@ def ensure_only_chart_is_modified(api_url):
     files_api_url = f'{api_url}/files'
     headers = {'Accept': 'application/vnd.github.v3+json'}
     r = requests.get(files_api_url, headers=headers)
-    pattern = re.compile(r"charts/(\w+)/([\w-]+)/([\w-]+)/([\w\.]+)/.*")
+    pattern = re.compile(r"charts/(\w+)/([\w-]+)/([\w-]+)/(.+?(?=\/))/.*")
     reportpattern = re.compile(r"charts/(\w+)/([\w-]+)/([\w-]+)/([\w\.]+)/report.yaml")
     count = 0
+    match_found = False
     for f in r.json():
-        if not pattern.match(f["filename"]):
-            print("PR should only modify chart related files")
+        filename = f["filename"]
+        match = pattern.match(filename)
+        if not match:
+            msg = f"[ERROR] PR should only modify chart related files: {filename}"
+            print(msg)
+            print(f"::set-output name=sanity-error-message::{msg}")
             sys.exit(1)
-        elif reportpattern.match(f["filename"]):
+        elif match:
+            if not match_found:
+                pattern_match = match
+                match_found = True
+
+        elif reportpattern.match(filename):
             print("[INFO] Report found")
             print("::set-output name=report-exists::true")
 
+    if match_found:
+        category, organization, chart, version = pattern_match.groups()
+        print("Downloading index.yaml", category, organization, chart, version)
+        r = requests.get(f'https://raw.githubusercontent.com/{repository}/{branch}/index.yaml')
+        if r.status_code == 200:
+            data = yaml.load(r.text, Loader=Loader)
+        else:
+            data = {"apiVersion": "v1",
+                "entries": {}}
+
+        crtentries = []
+        entry_name = f"{organization}-{chart}"
+        d = data["entries"].get(entry_name, [])
+        for v in d:
+            if v["version"] == version:
+                msg = f"[ERROR] Helm chart release already exists in the index.yaml: {version}"
+                print(msg)
+                print(f"::set-output name=sanity-error-message::{msg}")
+                sys.exit(1)
+
+        tag_name = f"{organization}-{chart}-{version}"
```

Comment:
tag_name is also determined in chartrepomanger.py, as a result it would be better to set tag_name as an output which chartrepomanger.py can use. This way we ensure both are using the same thing and we don't need to update two places if it ever changed for any reason. The same goes for entry_name.

Reply:
Done. bf4863c47c0f01ad2f4bfe80a388d432e6d621b9